### PR TITLE
cluster operator: init helm chart

### PR DESCRIPTION
**Pull request by zirain**
Signed-off-by: hejianpeng <hejianpeng2@huawei.com>

**What type of PR is this?**

**What this PR does / why we need it**:

/kind feature

**Which issue(s) this PR fixes**:
part of https://github.com/kurator-dev/kurator/issues/96

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cmd/cluster-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Cluster Operator
```
Comment (hzxuzhonghu) [inline cmd/cluster-operator/README.md:1]: This is not the right place to put readme

File: cmd/crd-gen-tool/main.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+func genAWS(outputDir string, version string) {
+	fmt.Printf("start to gen AWS crds, output: %s \n", outputDir)
+	infraComponentsYaml := fmt.Sprintf("https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/%s/infrastructure-components.yaml", version)
+	resp, err := http.Get(infraComponentsYaml)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c := newClient()
+	resources, err := c.Build(resp.Body, false)
```
Comment (hzxuzhonghu) [inline cmd/crd-gen-tool/main.go:103]: does not resp.Body need close?

File: hack/sync-crds.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+# gen crds
+rm -rf ${HELM_CRD_BASE}/*.x-k8s.io.yaml
+OUTPUT_DIR=${HELM_CRD_BASE} CLUSTER_API_PROVIDER_VERSION=${CLUSTER_API_PROVIDER_VERSION} AWS_PROVIDER_VERSION=${AWS_PROVIDER_VERSION} go run cmd/crd-gen-tool/main.go
+
+# sed cert
+sed -i "s|caBundle: Cg==|caBundle: ${CA_BUNDLE}|g" $(find ${HELM_CRD_BASE} -type f)
```
Comment (hzxuzhonghu) [inline hack/sync-crds.sh:33]: Is `Cg==` a placeholder?

**Review by hzxuzhonghu — APPROVED** [review]

File: manifests/charts/cluster-operator/templates/validatingwebhook.yaml [hunk 4]
```diff
@@ -0,0 +1,280 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/kurator-serving-cert
+  name: kurator-validating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1beta1
```
Comment (hzxuzhonghu) [inline manifests/charts/cluster-operator/templates/validatingwebhook.yaml:9]: use v1?

File: hack/sync-crds.sh (hunk 3 above)
Comment (zirain) [reply hack/sync-crds.sh:33]: yes

File: cmd/cluster-operator/README.md (hunk 1 above)
Comment (zirain) [reply cmd/cluster-operator/README.md:1]: will move to doc when stable

File: cmd/crd-gen-tool/main.go (hunk 2 above)
Comment (zirain) [reply cmd/crd-gen-tool/main.go:109]: done

**Pull request merged**